swupd/rename.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package swupd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+)
+
+// BSDIFFSIZE is smallest size of file to consider using bsdiff
+const BSDIFFSIZE = 200
+
+func renameDetection(manifest *Manifest, added []*File, removed []*File, c config) {
+	if len(added) == 0 || len(removed) == 0 {
+		return // nothing to rename
+	}
+	added = trimrenamed(added) // Make copies of input slices, tidy up whilst we are here
+	removed = trimrenamed(removed)
```

Comment:
I don't get the tidying up here. Both lists shouldn't have any DeltaPeer at this point. Maybe you just need a function to copy the slices (or copy them inline).

Reply:
Copy inline would work, I was just reusing something I had around....